src/actions/assignment.js

```diff
@@ -0,0 +1,67 @@
+import { browserHistory } from 'react-router';
+import fetch from 'isomorphic-fetch';
+import apiClient from 'panoptes-client/lib/api-client';
+
+import config from '../constants/config';
+import * as types from '../constants/actionTypes';
+
+
+// Action creators
+const { root, assignments } = config.eduAPI;
+
+export function createAssignment(assignment, classroomId) {
+
+  const classroomData = Object.assign({}, {
+      id: classroomId,
+      type: 'classrooms',
+    })
+  const metadata = Object.assign({}, {
+    classifications_target: assignment.classifications_target,
+    description: assignment.description,
+    duedate: assignment.duedate,
+  });
+  const studentData = assignment.students.map(student_id => { return {id: student_id, type: 'student_user'} });
+  const subjectData = assignment.subjects.map(subject_id => { return {id: subject, type: 'subjects'} });
+
+  return dispatch => {
+    const createAction = { ...assignment, type: types.CREATE_ASSIGNMENT };
```

Comment:
@rogerhutchings I think you used that pattern in `actions/teacher` for better readability (it's not reused there either). So I was trying to be consistent.